app/javascript/components/Pages/EmotionEntry.js

```diff
@@ -1,26 +1,136 @@
+import React, {Fragment, useState, useEffect} from 'react';
+import Form from 'react-bootstrap/Form';
+import {apiRequest} from "../requests/axios_requests";
+import {ShoutOutIcon, BtnBack, BtnNext, HelpIcon, Header, Wrapper} from "../UI/ShareContent";
+import iconNegative from "../../../assets/images/icon_negative.svg";
+import iconNeutral from "../../../assets/images/icon_neutral.svg";
+import iconPositive from "../../../assets/images/icon_positive.svg";
+import axios from "axios";
+import {CSSTransition} from 'react-transition-group';
+import {backHandling} from "../helpers/helpers";
+
+const EmotionEntry = ({data, setData, saveDataToDb, steps, service}) => {
 
-const EmotionEntry =  ({data, setData, saveDataToDb, steps, service}) => {
   const {isLoading, error} = service
+  const [emotion, setEmotion] = useState({ word: data.emotion?.word || '', category: data.emotion?.category || '' });
+  const [emotions, setEmotions] = useState([]);
+  const [show, setShow] = useState(false);
+
+  const onChangeEmotion = (e) => {
+    const { name, value } = e.target;
+    const trimmedValue = value.toLowerCase().trim();
+    setEmotion({ ...emotion, [name]: trimmedValue });
+
+    const existingEmotion = emotions.find(emotion => emotion.attributes.word === trimmedValue);
+    if (existingEmotion) {
+      setEmotion(prevState => ({ ...prevState, category: existingEmotion.attributes.category }));
+    }
+  };
+
+  const emojis = [
+    { name: 'negative', icon: <img className="icon-negative" src={iconNegative} alt={"negative"}/> },
+    { name: 'neutral', icon: <img className="icon-neutral" src={iconNeutral} alt={"neutral"}/> },
+    { name: 'positive', icon: <img className="icon-positive" src={iconPositive} alt={"positive"}/> },
+  ];
+  const [hoveredEmoji, setHoveredEmoji] = useState("");
+  const handleEmojiClick = (name) => {
+    setSelectedEmoji(name);
+    setEmotion(prevState => ({...prevState, category: name}));
+    setHoveredEmoji("");
+  };
 
   const handlingOnClickNext = () => {
-    steps.push('meme-selection')
-    saveDataToDb( steps, {})
-  }
-  //id:"1.1.", step:"emotion-entry"
-  return (
-    <Fragment>
-      { !!error && <p>{error.message}</p>}
-      { !isLoading && !error  &&
+    const dataFromServer = (word) =>{
+      steps.push('meme-selection')
+      saveDataToDb( steps, {emotion_id: word.data.id} )
+    }
+    const dataRequest = {
+      emotion: {word: emotion.word, category: emotion.category}
+    }
+    apiRequest("POST", dataRequest, dataFromServer, ()=>{}, "/api/v1/emotions").then(response => {
+    });
+  };
+
+  const [selectedEmoji, setSelectedEmoji] = useState("");
+  const getEmojiClass = (category, name) => {
+    if (category === "positive") {
+      return "input-positive";
+    } else if (category === "negative") {
+      return "input-negative";
+    } else if (category === "neutral") {
+      return "input-neutral";
+    } else {
+      return "";
+    }
+  };
```

Comment:
@Maryna-Harasko for big conditions in js more convenient use switch
https://medium.com/nerd-for-tech/a-case-to-switch-using-switch-statements-in-react-e83e01154f60